core/src/main/java/io/ultrabrew/metrics/Histogram.java

```diff
@@ -0,0 +1,20 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the Apache License 2.0 license. See LICENSE file in Ultrabrew Metrics
+// for terms.
+
+package io.ultrabrew.metrics;
+
+import io.ultrabrew.metrics.util.DistributionBucket;
+
+public class Histogram extends Gauge {
```

Comment:
There is no need to have a Histogram. Why is there a histogram in instrumentation. Histograms can be used for Gauge or Timer in the reporting side, but what is histogram really instrumenting? It does not make any sense in instrumentation side.

Reply:
We need to know the type of data that has to be maintained. ie, histogram vs gauge. if we want histogram a reporter concern, the information that the caller wants a histogram has to somehow be passed down. We could create a different metric/gauge registration api to support that.

Reply:
There are two distinct phases as per our readme and documentation, etc. One is to instrument code, i.e. I am measuring now this and that etc. How we then report it is a separate issue. I can use histograms, latency buckets, ngrams etc to report on the timer. I must not need to change the instrumentation code when I decide to switch this.

Reply:
Histograms, ngrams, etc require configuration which may be specific to a metric. In case of histograms, that's the bucket spec. Even if we treat Histograms as a reporter level concern, the instrumented code will have to pass the appropriate configurations to the reporter(s) separately from how metrics are registered.

One could argue that metrics should all be predefined in a single place and instrumented code should just use these metrics. I don't believe that'll be feasible for all applications.

We could go for code like this:

```
MetricRegistry metricRegistry = new MetricRegistry();

Reporter metricsReporter = new Reporter(...);
metricRegistry.addReporter(reporter);

Gauge cacheSizeGauge = metricRegistry.gauge("cacheSize");
Map<String,String> cache = new java.util.Map<>();
String[] tagList = new String[] { "host", hostName };
metricsReporter.addAggregator(cacheSizeGauge, Reporter.TYPE_HISTOGRAM, <spec>)

// ... do something

cacheSizeGauge.set(cache.size(), tagList);
```

Thoughts?

Reply:
Resolve conversation button really sucks.

The configuration is reporter specific anyway, you cant give the configuration in the instrumentation class, as you need to support multiple reporters and one might report it as p99 only and other as avg and 3rd as latency buckets. 

There was already ask in our discussion with smruti the need for the config overrides on class AND metric id level

Reply:
Are you thinking of providing overrides in a different class, away from where metric is originally registered? I'm not sure if that's the right approach. Configuration around a metric should be managed in a common place.. preferably where the metric is declared.

Reply:
That is against the whole design where instrumentation and reporting are separated

Reply:
They would still be separated. Specifically, this configuration would be away from where the metric is set/modified.

Reply:
No, it should be set where reporter is defined.